### PR TITLE
splunkhereceiver: update labels -> attributes

### PR DESCRIPTION
**Pull request by codeboten**
**Description:**
Following the changes in core to OTLP 0.9.0 to move away from `Labels` and on to `Attributes`

**Link to tracking Issue:** Part of https://github.com/open-telemetry/opentelemetry-collector/issues/3535

### CODE REVIEW

File: receiver/splunkhecreceiver/splunkhec_to_metricdata.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func buildLabels(dimensions map[string]interface{}) pdata.StringMap {
 			// TODO: Log or metric for this odd ball?
 			continue
 		}
-		labels.Insert(key, fmt.Sprintf("%v", val))
+		attributes.InsertString(key, fmt.Sprintf("%v", val))
```
Comment (anuraaga) [inline receiver/splunkhecreceiver/splunkhec_to_metricdata.go:146]: Should this check the type of `val` and insert the appropriate typed value? If it's too much for this PR can you file an issue and link it here?

**Review by tigrannajaryan — APPROVED** [review]
LGTM

**Review by atoulme — APPROVED** [review]
LGTM

**Review by dmitryax — APPROVED** [review]

File: receiver/splunkhecreceiver/splunkhec_to_metricdata.go (hunk 1 above)
Comment (tigrannajaryan) [reply receiver/splunkhecreceiver/splunkhec_to_metricdata.go:146]: @codeboten please file an issue.

**Pull request merged**